docs: explain csrf defenses

vinckr opened the pull request:
Adds explanation to the security / threat model document on how Ory Identities handles CSRF attacks.

aeneasr commented on docs/kratos/concepts/security.mdx:
This isn't correct unfortunately. CSRF can happen without a session too. One such attack is login csrf ( https://support.detectify.com/support/solutions/articles/48001048951-login-csrf )
. We deploy regular CSRF counter measures as explained here:

- https://owasp.org/www-community/attacks/csrf
- https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html

Instead of making the session cookie CSRF resistant, we have a dedicated CSRF cookie whose purpose it is to protect against CSRF, even if no session exists yet.

vinckr replied:
> Instead of making the session cookie CSRF resistant

We still use the [sameSite attribute](https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#samesite-cookie-attribute) as well though, right? 

Is the dedicated CSRF cookie what they describe in "[Double Submit](https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#double-submit-cookie)" cookie?
or what they describe under Login CSRF?

> Login CSRF can be mitigated by creating pre-sessions (sessions before a user is authenticated) and including tokens in login form. 

If it is really complex to describe here, maybe it is not worth it to go in such detail and just say there are multiple protections against multiple threat vectors regarding CSRF?

aeneasr replied:
No we use https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#synchronizer-token-pattern

aeneasr reviewed (changes requested):
See comment

vinckr commented:
Thanks for the clarification!


- Rewrote the CSRF text to clarify that we mainly use the anti-csrf cookie
- Removed the "What is CSRF" section from the troubleshooting doc and moved that content also to the Threat models doc

This way you can learn about CSRF and how Ory protects against it in the threat models doc and the troubleshooting is just focused on how to fix errors related to that.

Feel free to take another look :)

fbolton reviewed (approved):
Text and layout looks good to me.

vinckr commented:
Can you please take a quick look before we merge @aeneasr ?

aeneasr reviewed (approved):
Great!
